### PR TITLE
Fix bug in MarkDown to JSON conversion

### PR DESCRIPTION
**Pull request by parthsharma1996**
**Proposed changes**:
- Add a strip_comments() function in md_to_json.py which removes MarkDown
	comments from a line
- The function uses a regexp to detect and remove comments
- Call the function before processing each line of the file being read

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_nlu/utils/md_to_json.py [hunk 1]
```diff
@@ -15,10 +15,16 @@
 intent_regex = re.compile('##\s*intent:(.+)')
 synonym_regex = re.compile('##\s*synonym:(.+)')
 example_regex = re.compile('\s*-\s*(.+)')
+comment_regex = re.compile('\s*<!--.*-->\s*')
 
 INTENT_PARSING_STATE = "intent"
 SYNONYM_PARSING_STATE = "synonym"
 
+def strip_comments(comment_regex,text): 
+    """ Removes comments defined by the argument 'comment_regex' from the
```
Comment (tmbo) [inline rasa_nlu/utils/md_to_json.py]: no space at the beginning. Function comments should either be single line or multi line with a whitespace, e.g.
lets do ```"""Remove comments defined by `comment_regex` from `text`. """```

Comment (parthsharma1996) [reply rasa_nlu/utils/md_to_json.py]: Okay, so I should  1. fix it,  2. commit and then 3. push again, right ?
or is there a quicker way to do this?

**Comment by tmbo** [comment]
Thanks a lot :rocket:

**Comment by parthsharma1996** [comment]
@tmbo I've discovered a a few bugs/improvements in this pull request (unfortunately, _after_ the merge) 


1. The correct regex for comments would be `'<!--[\s\S]*?-->'`  instead of `'\s*<!--.*-->\s*'`
	1. Since  `.*` has a greedy behavior by default, the regex used would 
             

        1.  also remove incorrect comments like `'<!-- sdfad --> -->'` and won't throw any error.  
        2. remove the entire line along with the useful text, in the case of two comments in a single line like this 
         ```
         <!--first comment--> useful text <!--second comment-->
         ```

       The `?`  after `.*` ensures "lazy" behaviour.
   2. We shouldn't remove the white-spaces along with the comment ( the `\s*` in the re                    gexp) . Currently, the following line
          ```
         This is <!--first comment--> an example
          ```
          would be passed as `This isan example` instead of `This is an example`, which is unexpected behavior.
    3. The `[\s\S]` instead of `.` ensures that multi-line comments are also supported. Of course we'll have to apply to `strip_comments` function to the entire file instead of single lines as we're doing now.

2. All the regexps use literal strings instead of raw strings. Although it's not causing any problems, it seems like a good practice to use raw strings in python when dealing with regexps, as seen [here](https://stackoverflow.com/questions/12871066/what-exactly-is-a-raw-string-regex-and-how-can-you-use-it) and [here](https://docs.python.org/3/library/re.html).

Let me know if I should create another pull request to fix this. I'd be happy to do that.

**Comment by tmbo** [comment]
Yes - should have taken a closer look at the regex. Would be great if you could create another PR

**Comment by parthsharma1996** [comment]
Okay, on it. Will also write  tests this time

**Pull request merged**